src/ServiceControl.Transports.SQS/SQSTransportCustomization.cs

```diff
@@ -29,8 +30,17 @@ static void ConfigureTransport(TransportExtensions<SqsTransport> transport, Tran
         {
             var builder = new DbConnectionStringBuilder { ConnectionString = transportSettings.ConnectionString };
 
-            PromoteEnvironmentVariableFromConnectionString(builder, "AccessKeyId", "AWS_ACCESS_KEY_ID");
-            PromoteEnvironmentVariableFromConnectionString(builder, "SecretAccessKey", "AWS_SECRET_ACCESS_KEY");
+            if (builder.ContainsKey("AccessKeyId") || builder.ContainsKey("SecretAccessKey"))
```

Comment:
Given that we required either both or none should we make here both mandatory and thus change the bool condition to `&&`?

Reply:
yeah but if we make both mandatory then we'd fall into the else part and thus fall back to the else if one is missing. Then the SDK constructor would fail. Isn't that more aligned with the case if both are not provide IAM role either then also the SDK constructor would fail?

Reply:
Using else allows the exception they receive to be more specific than the generic auth error they will receive from the SDK constructor. The else will also log something about using environment variables or IAM roles, when they were attempting to provide the credentials in the connection string. Wouldn't this be more confusing?

Reply:
I'm ok to follow your logic. Let's move ahead with or